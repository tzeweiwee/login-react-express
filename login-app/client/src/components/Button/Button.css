.buttonContainer {
  margin: 5px;
}

.button {
  background-color: #222;
  width: 100px;
  height: 30px;
  box-sizing: border-box;
  border-color: #222;
  border-radius: 10px;
  font-size: 14px;
  color: white;
  cursor: pointer;
}

.button:hover {
  background-color: #4f4f4f;
  border-color: #4f4f4f;
}