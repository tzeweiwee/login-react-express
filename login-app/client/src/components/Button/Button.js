import React from 'react';
import PropTypes from 'prop-types';
import styles from './Button.css';

const Button = ({ type, text, onClick }) => {
    return (
        <div className={styles.buttonContainer}>
          <input className={styles.button} type={type} value={text} onClick={onClick} />
        </div>
    );
};

Button.defaultProps = {
    type: 'button'
};

Button.propTypes = {
  text: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  type: PropTypes.string
};

export default Button;