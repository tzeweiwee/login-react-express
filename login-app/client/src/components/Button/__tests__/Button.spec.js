import React from 'react';
import { shallow } from 'enzyme';
import Button from '../Button';

describe('Button', () => {
  const title = 'Button';
  it('should render Button component with text', () => {
      const wrapper = shallow(<Button text={title} onClick={() => {}}/>);
      expect(wrapper).toHaveLength(1);
  });

  it('should match Button snapshot', () => {
    const wrapper = shallow(<Button text={title} onClick={() => {}}/>);
    expect(wrapper).toMatchSnapshot();
});
});
