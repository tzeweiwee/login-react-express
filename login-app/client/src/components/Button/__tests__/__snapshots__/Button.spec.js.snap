// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match Button snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    onClick={[Function]}
    text="Button"
    type="button"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <input
        className={undefined}
        onClick={[Function]}
        type="button"
        value="Button"
/>,
      "className": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": undefined,
        "onClick": [Function],
        "type": "button",
        "value": "Button",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <input
          className={undefined}
          onClick={[Function]}
          type="button"
          value="Button"
/>,
        "className": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": undefined,
          "onClick": [Function],
          "type": "button",
          "value": "Button",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
